src/components/validation/resources/event.rs

```diff
@@ -59,6 +72,9 @@ pub enum TestEvent {
     /// The event is used, as-is, without modification.
     Passthrough(Event),
 
+    /// The event is used, as-is, without modification by the framework, but we expect it to fail.
+    PassthroughFail(Event),
+
```

Comment:
I don't understand why this new variant exists.

I see it being used, but I don't see any additional/diverging logic that occurs when this is the variant used.

Reply:
I see what you are seeing. On the surface it looks like we could just use `Modified` instead. I'm guessing the main reason I added this variant is because the doc comment for Modified says:

> this mode is made available for when a test case wants to exercise the failure path, but cannot cause a failure simply by constructing the event in a certain way i.e. adding an invalid field

, which is exactly what the new variant represents.

Moreover, I think the code path for `Modified`, results in us doing stuff which would cause a different type of failure than what we need.

Also in the doc comment for `Modified` :

>  For transforms and sinks, generally, the only way to cause an error is if the event itself is malformed in some way, which can be achieved without this test event variant.

Reply:
Right, so perhaps one thing I would suggest changing is actually updating `Modified` to allow configuring _how_ things are modified.

I don't want to pigeonhole the design by listing things out, but, essentially: `Modified` was always a crappy way to describe the intent of the variant, and really what it should have been was a way to say "here's an event, and here's how I want you to potentially modify it compared to the default behavior, which should cause it to fail".

This probably speaks to the nature of the enum itself, and the variants needing to be named to actually describe the intent: this event should succeed, this event should fail (and here's the modification to make it fail), etc.

Reply:
Ok, I've made adjustments to both enums to make things a little more clear and purpose driven